main.cpp

```diff
@@ -13,27 +14,27 @@ int main(int argc, char *argv[])
     try
     {
 
-		BaseUnit Unit1 = BaseUnit::parseUnit(argv[1]);
-		BaseUnit Unit2 = BaseUnit::parseUnit(argv[2]);
+		Player Unit1 = Player::parsePlayer(argv[1]);
+		BaseUnit Unit2 = Player::parsePlayer(argv[2]);
```

Comment:
Itt inkább a `BaseUnit::parseUnit`ot kellene meghívni, mert most egy `Player`t próbál eltárolni egy `BaseUnit` változóban. Hibát nem okoz, mert az extra tagváltozókat levágja, de azért nem szép. Csodálom is, hogy nem ír rá warningot a fordító.